src/App/App.jsx

```diff
@@ -8,34 +8,32 @@ import { AuthContext } from '../Hooks/useAuth';
 import { Login } from '../components/Pages/Login/Login';
 import style from './App.module.css';
 import { getRoute } from '../Routes/getRoute';
+import { About } from '../components/Pages/About/About';
+import { Presentation } from '../Presentation/Presentation';
 
-export class App extends React.PureComponent {
-  render() {
-    return (
-      <Container className={style.container}>
-        <AuthContext.Consumer>
-          {({ isAuth, role }) => {
-            return (
-              <ErrorBoundary>
-                {isAuth ? (
-                  <>
-                    <Header />
-                    <main>{getRoute(role)}</main>
-                  </>
-                ) : (
-                  <>
-                    <Redirect to='/login' />
-                    <main>
-                      <Login />
-                    </main>
-                  </>
-                )}
-              </ErrorBoundary>
-            );
-          }}
-        </AuthContext.Consumer>
-        <Footer />
-      </Container>
-    );
-  }
+export function App() {
+  const { isAuth, role } = useContext(AuthContext);
+
+  return (
+    <Container className={style.container}>
+      <Header />
+      <ErrorBoundary>
+        {isAuth ? (
+          <main>{getRoute(role)}</main>
```

Comment:
I understand what you mean, but I can't do it because I get dependency loop errors. For example I use `Сontext` in `Trasks` -> `Trasks` in `MembRouts` -> `MembRouts` in  `Context`. And so in several places.